plugins/generated-creature-renamer.cpp

```diff
@@ -0,0 +1,159 @@
+#include "Console.h"
+#include "Core.h"
+#include "DataDefs.h"
+#include "Export.h"
+#include "PluginManager.h"
+#include "df/world.h"
+#include "df/world_raws.h"
+#include "df/creature_raw.h"
+#include "df/caste_raw.h"
+#include "modules/World.h"
+
+//#include "df/world.h"
+
+using namespace DFHack;
+
+DFHACK_PLUGIN("generated-creature-renamer");
+REQUIRE_GLOBAL(world);
+
+command_result list_creatures(color_ostream &out, std::vector <std::string> & parameters);
+
+DFhackCExport command_result plugin_init(color_ostream &out, std::vector <PluginCommand> &commands)
+{
+    commands.push_back(PluginCommand(
+        "list-generated",
+        "Prints a list of generated creature tokens. Use \"list-generated detailed\" to show descriptions.",
+        list_creatures,
+        false, //allow non-interactive use
+        "Prints a list of generated creature tokens. Use \"list-generated detailed\" to show descriptions."
+    ));
+    return CR_OK;
+}
+
+DFhackCExport command_result plugin_shutdown(color_ostream &out)
+{
+    return CR_OK;
+}
+
+DFHACK_PLUGIN_IS_ENABLED(is_enabled);
+
+DFhackCExport command_result plugin_enable(color_ostream &out, bool enable)
+{
+    is_enabled = enable;
+    return CR_OK;
+}
+
+#define NUM_DESC 227
+std::string descriptors[NUM_DESC] = { " blob", " quadruped", " humanoid", " silverfish", " mayfly", " dragonfly",
```

Comment:
Oh, this is searching the description. Makes sense, although you could probably prepend the space later (instead of relying on it to be present in every entry here), e.g.
```c++
size_t pos = creatureRaw->caste[0]->description.find(" " + descriptors[j]);
```

I would probably also use an STL container instead of keeping track of the size manually, but it's not a big deal.